### ISSUE TITLE
Metadata: Bearer Token Format

### ISSUE THREAD

**Issue by mmccool** [w3c/wot-security#80]
OpenAPI does not specify the terms used to identify different kinds of bearer tokens, since they are not created by the client, but by an authentication server. Should we be stricter, or not?

**Comment by ereshetova** [comment]
I don't think it is the WoT job to define details for bearer tokens and other details since WoT is 1) likely not going to do any verification by itself (my understanding that this task is going to be carried by underneath protocol bindings) 2) all these formats and details might change (including the set of recommended and not recommended ones) and we probably don't want to follow up constantly all the standards for this.

**Comment by mmccool** [comment]
Yes, that makes sense.  What matters is the communication pattern, not the content of the payload.  Treating the internals of the token as opaque seems like the safest approach.

**Comment by mmccool** [comment]
See PR https://github.com/w3c/wot-security/pull/86
I thought about this some more and I think we need to distinguish tokens (which do have internal structure we can access) and API keys (which are opaque).   An example of useful internal data in a token might include an expiry date or the identity of the user, for instance.   To be useful, we also need to specify the format of the token.  My revised proposal takes this into account...

**Comment by mkovatsc** [comment]
We should not go too deep with the description. What fields exist and how to use them must be handled by the security implementation that was identified by `scheme`. Within a scheme, specific parameters should inform about implementation requirements, e.g., specific algorithms (RS256 vs HS256) or token formats (e.g., opaque vs JWT).

I don't understand your split into schemes `Bearer` and `OAuth2`.

I am not sure how `OAuth2` will relate to Proof-of-Possession, i.e., do we need a parameter or a new scheme.

**Comment by mmccool** [comment]
I'm assuming tokens might be used on their own, i.e. not always in combination with OAuth2.   In that case a token type can both be a parameter of the OAuth2 scheme and "primitive" schemes in their own right.   I do need to look more into this, though.

**Comment by mkovatsc** [comment]
When we choose `OAuth2` as `scheme`, the `as` (authorization server) field could be optional. In that case, the client needs to know on its own how to acquire a Token -- for instance through that manual "on their own" way as you describe above.

In that case, the scheme `OAuth2` needs a field for the token_type with the two common values `Bearer` and `PoP`, although some systems might use their custom token types, i.e., some custom value.

I am not sure about how to describe JWT or not JTW, and if CWT is even an option for OAuth2.0...

**Comment by mmccool** [comment]
Add an explicit test for this in the plugfest planning document, then close this issue.   We have metadata for this... what we need to test is whether we have the *right* metadata.
